### PR TITLE
tutorial: adding tutorial from scale18x originally written by @conejoninja

### PR DESCRIPTION
**Pull request by deadprogram**
This PR adds the tutorial from scale18x originally written by @conejoninja

It can hopefully serve as a starting point once it has been updated a little for the newest TinyGo, hence I am marking at as draft for the moment.

### CODE REVIEW

File: tutorial/README.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
+# GoBadge Tutorial
+
+## What you need
+
+    - GoBadge aka Adafruit PyBadge
+    - Personal computer with Go 1.18 and TinyGo installed, and a serial port.
```
Comment (conejoninja) [inline tutorial/README.md]: Latest go version is 1.19.1
Latest TinyGo version is 0.25.0. 

We should at least recommend 0.25.0 for the rp2040 flash issue that happened at gophercon.eu

File: tutorial/README.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+## Installation
+
+### Go
+
+If somehow you have not installed Go on your computer already, you can download it here:
+
+https://golang.org/dl/
```
Comment (conejoninja) [inline tutorial/README.md]: New download URL is : https://go.dev/dl/

File: tutorial/snake/main.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	})
+	display = st7735.New(machine.SPI1, machine.TFT_RST, machine.TFT_DC, machine.TFT_CS, machine.TFT_LITE)
+	display.Configure(st7735.Config{
+		Rotation: st7735.ROTATION_90,
+	})
+
+	buttons = shifter.NewButtons()
+	buttons.Configure()
+
+	snakeGame := Game{
```
Comment (conejoninja) [inline tutorial/snake/main.go]: Copy improvements from this fix :  https://github.com/tinygo-org/gobadge/commit/0f6354a46bc8154872437c7a7e01bd50b7f97d5b

**Review by conejoninja — COMMENTED** [review]
I wasn't able to run the code, but I guess it should work fine or with minor tweaks (à la snake fix)

**Comment by conejoninja** [comment]
Additionally, this needs to be changed in order to match the edgeBadge colors/position : https://github.com/tinygo-org/gobadge/blob/main/menu.go#L24-L31 but it's probably out of reach for this PR

File: tutorial/snake/main.go (hunk 3 above)
Comment (deadprogram) [reply tutorial/snake/main.go]: done

File: tutorial/README.md (hunk 1 above)
Comment (deadprogram) [reply tutorial/README.md]: done

File: tutorial/README.md (hunk 2 above)
Comment (deadprogram) [reply tutorial/README.md]: done

**Comment by deadprogram** [comment]
All updates made so at least the tutorial all functions. :smile_cat:

**Comment by conejoninja** [comment]
Thanks for your work!

**Pull request merged**